src/main/java/com/github/shyiko/mysql/binlog/event/deserialization/ColumnType.java

```diff
@@ -67,17 +67,17 @@ public int getCode() {
         return code;
     }
 
-    private static final Map<Integer, ColumnType> INDEX_BY_CODE;
+    private static final ColumnType[] INDEX_BY_CODE;
 
     static {
-        INDEX_BY_CODE = new HashMap<Integer, ColumnType>();
+        INDEX_BY_CODE = new ColumnType[ColumnType.GEOMETRY.code + 1];
```

Comment:
What was the purpose of this change? Isn't this array going to have a big hole in the middle between TIME_V2 and JSON?

Reply:
There will be a hole in the middle but `byCode` is on a hot path for deserialization (and connector segment) so bypass `Map` lookup makes a significant difference. Considering that array length is only 256 elements it is a nice trade off.

Reply:
I also tested HashMap and array layout in memory and it seems that HashMap memory footprint is even larger.